update docs for phased execution (#102)
Green-Software-Foundation/if-docs: docs/major-concepts/manifest-file.md

Each component has some configuration, some input data, and a plugin pipeline.

- `pipeline`: a list of plugins that should be executed for a specific component. This is broken down into three subsections representing distinct phases of execution that can be triggered independently using command line flags. These subsections are:

>   - observe: the plugins that generate input data

Reviewer: ```suggestion
  - `observe`: the plugins that generate input data
```

>   - regroup: configuration for regrouping input data by given keys

Reviewer: ```suggestion
  - `regroup`: configuration for regrouping input data by given keys
```

>   - compute: the plugins that operate over input data and generate output data

Reviewer: ```suggestion
  - `compute`: the plugins that operate over input data and generate output data
```

The separation of timestamps in the `inputs` array determines the temporal granularity of your impact calculations. The more frequent your observations, the more accurate your impact assessment.

## Creating input data

> The plugins in the `observe` part of the pipeline generate `input` data. The manifest file should not have `input` data when the `observe` phase is executed. Plugins in this phase *only* generate input data, they can never generate output data. If you run the opbserve phase on its own (by runnin `if-run --observe`) then your manifest will be returned populated with input data according t the plugins you included in your `observe` pipeline.

Reviewer 2: ```suggestion
The plugins in the `observe` part of the pipeline generate `input` data. The manifest file should not have `input` data when the `observe` phase is executed. Plugins in this phase *only* generate input data, they can never generate output data. If you run the observe phase on its own (by running `if-run --observe`) then your manifest will be returned populated with input data according to the plugins you included in your `observe` pipeline.
```

## Regrouping a manifest file

> The second phase of manifest execution is `regroup`. This reorganizes existing `input` data into a new structure using keys provided in the `regroup` config in the manifest. For example, a mnaifest with the following `tree`:

Reviewer 2: ```suggestion
The second phase of manifest execution is `regroup`. This reorganizes existing `input` data into a new structure using keys provided in the `regroup` config in the manifest. For example, a manifest with the following `tree`:
```

                  duration: 300
                  cloud/instance-type: B1
                  cloud/region: uk-west
                  cpu/utilization: 67
                - timestamp: 2023-07-06T10:00
                  duration: 300
                  cloud/instance-type: B1
                  cloud/region: uk-west
                  cpu/utilization: 1
            uk-east:
              inputs:
                - timestamp: 2023-07-06T00:00
                  duration: 300
                  cloud/instance-type: B1
                  cloud/region: uk-east
                  cpu/utilization: 11
                - timestamp: 2023-07-06T05:00
                  duration: 300
                  cloud/instance-type: B1
                  cloud/region: uk-east
                  cpu/utilization: 67
                - timestamp: 2023-07-06T10:00
                  duration: 300
                  cloud/instance-type: B1
                  cloud/region: uk-east
                  cpu/utilization: 1
```

## Computing a manifest file

Impact Framework computes manifest files. For each component in the tree, the `inputs` array is passed to each plugin in the `compute` pipeline in sequence.

> In order for the `compute` pahse to execute correctly, the manifest nmeeds to have `input` data available. 

Reviewer 2: ```suggestion
In order for the `compute` phase to execute correctly, the manifest needs to have `input` data available. 
```

Each plugin _enriches_ the `inputs` array in some specific way, typically by adding a new `key-value` pair to each observation in the array. For example, the `teads-curve` plugin takes in CPU utilization expressed as a percentage as an input and appends `cpu/energy` expressed in kWh. `cpu/energy` is then available to be passed as an input to, for example, the `sci-e` plugin.

This implies a sequence of plugins where the inputs for some plugins must either be present in the original manifest file or be outputs of the preceding plugins in the pipeline.

There are also plugins and built-in features that can synchronize time series of `observations` across an entire tree and aggregate data across time or across components.

## Running combinations of phases

> It is possible to run each phase of the execution individually, or together. You can choose to *only* run the `observe`, `regroup` ro `compute` phases of the manifest execution. This saves you from having to re-execute entire manifests every time you want to tweak something, making it a greener way to use IF.

Reviewer 2: ```suggestion
It is possible to run each phase of the execution individually, or together. You can choose to *only* run the `observe`, `regroup` or `compute` phases of the manifest execution. This saves you from having to re-execute entire manifests every time you want to tweak something, making it a greener way to use IF.
```
